mne/decoding/transformer.py

```diff
@@ -61,29 +87,51 @@ def fit(self, epochs_data, y):
         self : instance of Scaler
             Returns the modified instance.
         """
+        from sklearn.base import clone
+        from sklearn.preprocessing.data import _handle_zeros_in_scale
         if not isinstance(epochs_data, np.ndarray):
             raise ValueError("epochs_data should be of type ndarray (got %s)."
                              % type(epochs_data))
 
         X = np.atleast_3d(epochs_data)
-
-        picks_list = dict()
-        picks_list['mag'] = pick_types(self.info, meg='mag', ref_meg=False,
-                                       exclude='bads')
-        picks_list['grad'] = pick_types(self.info, meg='grad', ref_meg=False,
-                                        exclude='bads')
-        picks_list['eeg'] = pick_types(self.info, eeg=True, ref_meg=False,
-                                       meg=False, exclude='bads')
-
-        self.picks_list_ = picks_list
-
-        for key, this_pick in picks_list.items():
-            if self.with_mean:
-                ch_mean = X[:, this_pick, :].mean(axis=1)[:, None, :]
-                self.ch_mean_[key] = ch_mean  # TODO rename attribute
-            if self.with_std:
-                ch_std = X[:, this_pick, :].mean(axis=1)[:, None, :]
-                self.std_[key] = ch_std  # TODO rename attribute
+        n_epochs, n_channels, n_times = X.shape
+        X = X.transpose(1, 0, 2)
+        self.coef_ = np.zeros(n_channels)
+
+        if self.scalings == 'standard':
+            self.ch_mean_ = list()
+            self.std_ = list()
+            for _, pick in self._pick_types:
+                if self.with_mean:
+                    ch_mean = X[pick].mean(axis=1, keepdims=True)
+                    ch_mean = _handle_zeros_in_scale(ch_mean, copy=False)
+                    self.ch_mean_.append(ch_mean)  # TODO rename attribute
+                    self.coef_.flat[pick] = ch_mean
+                if self.with_std:
+                    ch_std = X[pick].mean(axis=1, keepdims=True)
+                    ch_std = _handle_zeros_in_scale(ch_std, copy=False)
+                    self.std_.append(ch_std)  # TODO rename attribute
+                    self.coef_.flat[pick] = ch_std
+
+        elif isinstance(self.scalings, dict):
+            for typ, pick in self._pick_types:
+                if typ in self.scalings:
+                    self.coef_[pick] = self.scalings[typ]
+
+        elif not isinstance(self.scalings, str):
+            X = X.reshape(n_channels, -1)
+            self.scalers_ = list()
+            for _, pick in self._pick_types:
+                scaler = clone(self.scalings)
+                scaler.fit(X[pick].T)
+                self.scalers_.append(scaler)
+                self.coef_.flat[pick] = scaler.scale_
+
+        elif self.scalings == 'default':
```

Comment:
Move this elif to first position, it'll be easier to read (if specific_string, elif specific_string, elif generic_string, elif dict, else error)